ServiceApp/src/org/droidplanner/services/android/core/mission/Mission.java

```diff
@@ -331,18 +327,22 @@ public void sendMissionToAPM() {
 
     private void updateComponentItems(){
         List<msg_mission_item> msgMissionItems = getMsgMissionItems();
-        msgMissionItems.remove(0);
         updateComponentItems(msgMissionItems);
     }
 
     private void updateComponentItems(List<msg_mission_item> msgMissionItems) {
+        if(msgMissionItems == null)
```

Comment:
@TShapinsky you should still clear the `componentItems` field.
Along the check for `null`, also check if the `msgMissionItems` is non-empty.